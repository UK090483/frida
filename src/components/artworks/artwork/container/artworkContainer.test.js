import React from "react"
import renderer from "react-test-renderer"
import ArtworkContainer from "./artworkContainer"
import { cleanup, fireEvent, render } from "@testing-library/react"
import "jest-styled-components"

jest.mock("../../../generic/Mouse/mouseRemote")

const onClick = jest.fn()
describe("ArtworkContainer ", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ArtworkContainer onClick={onClick}></ArtworkContainer>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("calls onClick", () => {
    const { getByRole, debug } = render(
      <ArtworkContainer onClick={onClick}>
        <h1>bla</h1>
      </ArtworkContainer>
    )

    fireEvent.click(getByRole("button"))
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
