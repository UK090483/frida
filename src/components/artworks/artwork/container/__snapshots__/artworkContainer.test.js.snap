// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtworkContainer  renders correctly 1`] = `
.c0 {
  width: 100%;
  background-clip: padding-box;
  margin-bottom: 80px;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="button"
/>
`;
